mtags/src/main/scala/scala/meta/internal/pc/TypeDefinitionProvider.scala

```diff
@@ -0,0 +1,152 @@
+package scala.meta.internal.pc
+
+import org.eclipse.{lsp4j => l}
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.pc.OffsetParams
+import scala.meta.internal.jdk.CollectionConverters._
+import scala.meta.internal.metals.CompilerOffsetParams
+
+class TypeDefinitionProvider(val compiler: MetalsGlobal, params: OffsetParams) {
+  import compiler._
+
+  def typedTree: Option[(RichCompilationUnit, Position, Tree)] = {
```

Comment:
Do we need all 3 as return values?  Those can be created separately when needed and just have the Tree returned here.